.github/workflows/deploy.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish Image
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log into ghcr
+        uses: docker/login-action@v2.1.0
+        with:
```

Comment:
Maybe a condition if it's a release same as deleted deploy.yaml.

Reply:
Make sense